test/test_sort.py

```diff
@@ -514,6 +519,24 @@ def test_field_present_in_some_items(self):
         # items without field last
         assert [i.id for i in results_desc] == [ids[2], ids[1], ids[0], ids[3]]
 
+    def test_int_field_present_in_some_items(self):
+        """Test ordering by a field not present on all items."""
+        # append int-valued 'foo' to two items (1,2)
+        Item._types = {"foo": types.Integer()}
```

Comment:
Instead of overwriting global `Item` try patching it:

```
    @patch("beets.library.Item._types", {"foo": types.Integer()})
    def test_int_field_present_in_some_items(self):
```

Then you can remove the `tearDown` that got added in line 475.

Reply:
Nice, thank you. I copied this pattern from some tests in `test_query.py` so I updated those to patch as well, for consistency's sake.